### DISCUSSION TITLE
Is dispatch! an OK alternative to alt + cut_err?

### DISCUSSION THREAD

**Discussion by adamchalmers** [winnow-rs/winnow#338] [Q&A] [answered]
Winnow tutorial [chapter 6](https://docs.rs/winnow/latest/winnow/_tutorial/chapter_6/index.html) suggests using `cut_err` in an `alt` if you know that each branch has a unique prefix and therefore shouldn't backtrack.

But isn't this also the use-case for `dispatch!` -- so you should consider using `dispatch!` in this specific case?

If so, maybe the chapter 6 tutorial should suggest this.

**Comment by epage** [comment, marked as answer]
In that exact case, yes, `dispatch` will get the job done and this is covered in [an earlier chapter](https://docs.rs/winnow/latest/winnow/_tutorial/chapter_3/index.html).  The point here is more to illustrate the concept of `cut_err`.  There are times when you can't convert a `alt` + `cut_err` into a `dispatch`.

I would be interested in better understanding your thought process that led to this to better understand how much of a problem it is.  I'm liking keeping the examples building on each other and I dislike having a lot of side notes for exception cases, so I'd lean towards keeping it as-is but someting led you to creating this.

**Comment by adamchalmers** [comment]
It's definitely not a problem, and once I properly understood cut_err it became clear to me that these two should be equivalent. I was mostly filing this to get your feedback on whether I should add a little note in that doc saying "in this specific case, you could use dispatch! to achieve the same effect, because dispatch will not try other branches". 

(I guess I like when docs link back to other related concepts but maybe you'd like to avoid little notes like that in order to keep the page tightly-focused on just cut_err)
